### PR TITLE
Load SSL cert if environment variables exist

### PR DESCRIPTION
**Pull request by craineum**
The following environment variables are inspected:

If `HTTPS` is set to true and `SSL_KEY_PATH` && `SSL_CERT_PATH` are set, we try to load the certs via webpack devServer.

This was needed to get past browser _non-trusted_ self signed certs to have a nice happy green browser experience. See more here: https://github.com/facebookincubator/create-react-app/issues/2759#issuecomment-316477866

### CODE REVIEW

File: packages/react-scripts/config/webpackDevServer.config.js [hunk 1]
```diff
@@ -10,13 +10,35 @@
...
+  if (https === 'true') {
+    if (key_path && cert_path) {
+      return { key: sslPath(key_path), cert: sslPath(cert_path) };
+    } else {
+      return true;
+    }
+  } else {
+    return false;
+  }
+})(process.env.HTTPS, process.env.SSL_KEY_PATH, process.env.SSL_CERT_PATH);
```
Comment (randycoulman) [inline packages/react-scripts/config/webpackDevServer.config.js:41]: Curious why you're using an IIFE here.  I like the way the code looks, but just curious about the decision to go this way.

File: packages/react-scripts/config/webpackDevServer.config.js [hunk 2]
```diff
@@ -10,13 +10,35 @@
...
+const fs = require('fs');
 const errorOverlayMiddleware = require('react-error-overlay/middleware');
 const noopServiceWorkerMiddleware = require('react-dev-utils/noopServiceWorkerMiddleware');
 const config = require('./webpack.config.dev');
 const paths = require('./paths');
 
-const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
 const host = process.env.HOST || '0.0.0.0';
+// ZEAL: ssl file loading
+const sslPath = function(path) {
```
Comment (randycoulman) [inline packages/react-scripts/config/webpackDevServer.config.js]: The name `sslPath` doesn't communicate to me what this is doing.  Maybe `readSslFile` or something?

**Review by randycoulman — APPROVED** [review]
A question and a (non-blocking) suggestion.  👍  for offering to do a PR upstream (I found your comment when looking into why this PR was necessary).  Might be worth linking to that issue from here for future reference.

Overall, LGTM!

File: packages/react-scripts/config/webpackDevServer.config.js (hunk 1 above)
Comment (craineum) [reply packages/react-scripts/config/webpackDevServer.config.js:41]: I didn't want to call the function down below. I really wanted these methods to be in react-dev-utils and then pass in the results to this method from `start.js` as variable name `https`. Maybe that is too preemptive here.

Comment (randycoulman) [reply packages/react-scripts/config/webpackDevServer.config.js:41]: The problem is that `start.js` is not the only place that this config is imported, and the other importer doesn't have the necessary context, so what's here is good.

**Pull request merged**